schema.go

```diff
@@ -87,10 +89,16 @@ func (rs *RootSchema) UnmarshalJSON(data []byte) error {
 		if sch, ok := elem.(*Schema); ok {
 			if sch.ID != "" {
 				ids[sch.ID] = sch
```

Comment:
@b5 I have one doubt, when `$id` does not have `#` (i.e. `/properties/firstName`), an entry is made in `ids` at `/properties/firstName` (the value of sch.ID), when `$id` has `#`, the entry will be made at `#/properties/firstName`, just pointing this out because I am not aware of core workings and what repercussions it may cause though I feel that if `$id` has `#`, we just remove `#` and then make entry to `ids` (ie. `/properties/firstName` instead of `#/properties/firstName`), please pardon me if this is a silly suggestion, I'm very new to this project.

Reply:
@daveamit an `$id` of `/properties/firstName` is an absolute path (e.g. filesystem path, which would reference a completely separate document), while an `$id` of `#/properties/firstName` is a fragment using JSON Pointer syntax (which references a location within the same document).  For exhaustive detail see RFCs 3986 and 6901.

Reply:
@handrews thanks for the explanation.